File: fixtures/scale/python/large/cipherlab/checks/check_veilmask.py
from cipherlab.ciphers.veilmask import veilmask_encrypt, veilmask_decrypt
from cipherlab.specs.veilmask_spec import veilmask_spec
from cipherlab.corpus.corpus_mask import corpus_mask


def check_veilmask():
    spec = veilmask_spec()
    for text in corpus_mask():
        sealed = veilmask_encrypt(text, spec["key"])
        opened = veilmask_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "mask"

File: fixtures/scale/ruby/large/lib/check_veilmask.rb
require_relative 'veilmask_encrypt'
require_relative 'veilmask_decrypt'
require_relative 'corpus_mask'

def check_veilmask
  corpus_mask.each do |text|
    sealed = veilmask_encrypt(text, 4)
    opened = veilmask_decrypt(sealed, 4)
    return false if opened != text
  end
  true
end
